### ISSUE TITLE
 polymorphic filter not working

### ISSUE THREAD

**Issue by Tallimum** [tom5454/Toms-Storage#459]
The  polymorphic filter is not working.

Using the following:
- Minecraft version 1.21
- Fabric version  15.11
- Tom's version 2.0.9

Behavior: Large connected inventory of double chests.  Each chest has an inventory connector cable with a filter.  Normal filter w/ 9 items works fine.  Tag filter works fine.  As soon as I put in the polymorphic filter I cannot insert items.  I left one item in and tied to insert  via the crafting terminal and it went into my dump chest.  In addition, none of the items in the chest are visible with the filter in place.  Take the filter out and I see all the items.

**Comment by ElSantoSuper** [comment]
I am experiencing this as well, using Tom's Storage 2.0.10, Minecraft 1.21, and Fabric 0.16.5.

After some testing, I can say that they specifically do not work in inventory connectors or cable connectors. They **do** work if assigned to an actual storage block (chest or barrel).

**Comment by DPlll** [comment]
Yep they never work for me. It's my first time using this mod and its fantastic but the polymorphic filters just don't work for some reason on this new version. I have made an entire item system on the regular filters and its taken hours just to program the regular item filter (which thankfully work) for the connectors and chests. Thank and i am limited to only 9 items per chest with the current filters which is making my life a living hell lol trying to figure this out. I think i might be able to use proxy blocks to connect the chest to up to 4 connectors and thus 4 filters but that'd be very expensive and I'm 50/50 on if that'll even work.

**Comment by DPlll** [comment]
extending the face of the chests doesn't make it work either so your strictly capped to 9 items per chest i believe unless tagged and do more

**Comment by MetalPaw** [comment]
I am getting the exact same issue as the main post - same scenario, same issue.

An even simpler setup to replicate, 3 output chests and 1 input chest. Each Inventory Cable Connector has a Polymorphic Filter in it. No other settings changed. Each output chest has a single item in it, each different from the others. Input chest has a stack of each of these items. No items go through the system.
Moving the filter to the chest starts filtering all items into it, regardless of whether it was in the chest already or not.

<img width="1920" height="1009" alt="Image" src="https://github.com/user-attachments/assets/cf22059b-4348-494a-a11b-93b4d0a7ccfe" />

**Comment by ssMatthyss** [comment]
Alright, so polymorphic filter applied on chest works, but only if the chest are connected via inventory trim and not the cables with inventory cable connector, then it breaks and doesn't work at all. Could it be fixed please.

I'm playing on 1.21.1, with Tom's Storage 1.21-2.3.0 Fabric

**Comment by Abraxas161** [comment]
Spent 8 hours lookin' to solve my way through this issue. Hadn't worked until a double blind involving Zniezek's method, to my immediate and thorough relief, began working. 

Through connectors, items will post to connected inventory that was connected first / for the longest amount of time. 

Through trim, items post normally. Fix the mod when you can, pretty please.
